app/models/event.rb

```diff
@@ -22,6 +22,25 @@ class Event < ActiveRecord::Base
   has_many :date_ranges
   accepts_nested_attributes_for :date_ranges
 
+  # Setter for max_participants
+  # @param [Int Float] the max number of participants for the event or infinity if it is not limited
+  # @return none
+  def max_participants=(value)
+    if value == Float::INFINITY
+      self[:participants_are_unlimited] = true
```

Comment:
Because max_participants is an integer
But integers don't automatically support infinity, since they have no values that could be used (floats have values like 1/0 that will probably not be used for anything, those are then used for stuff like infinity)